lib/sycamore/sycamore/query/strategy.py

```diff
@@ -229,6 +229,70 @@ def __call__(self, plan: LogicalPlan, **kwargs) -> LogicalPlan:
             del plan.nodes[plan.result_node]
             plan.result_node = penultimate
             n = plan.nodes[plan.result_node]
+        return LogicalPlan.model_validate(plan)
+
+
+class LimitLlmOperations(LogicalPlanProcessor):
+    """
+    Add a limit node before certain operators. This is useful to make some queries faster, although
+    less accurate.
+
+    Args:
+        types: List of node types to limit; e.g. "LlmFilter", "SummarizeData"
+        limit: How many documents to limit to
+        message: Optional message to add to the description of the added limit node. Default is
+            "Limit the number of documents going through the following llm operation for interactivity"
+    """
+
+    def __init__(self, types: list[str], limit: int, message: Optional[str] = None):
+        self._types = types
+        self._limit = limit
+        self._message = (
+            message or "Limit the number of documents going through the following llm operation for interactivity"
+        )
+
+    def __call__(self, plan: LogicalPlan, **kwargs) -> LogicalPlan:
+        new_nodes: dict[int, Node] = {}
+        new_nodes_i = 0
+        old_to_new_mapping: dict[int, int] = {}
+        for old_i in sorted(plan.nodes.keys()):
+            node = plan.nodes[old_i]
+            new_inputs = [old_to_new_mapping[inp] for inp in node.inputs]
+            # Don't add a limit after another limit. Note - this is close to but not correct.
```

Comment:
`# TODO: Fix this to set the limit to min(explicit limit, applied limit).  ... (rest of comment)`

On second consideration, why not just fix this?  In the case of a single predecessor limit node, it should just be `new_nodes[new_inputs[0]].num_records = min(...)`
You can leave the TODO on multiple predecessors and some have a limit since that won't happen right now. Probably emit a warning "unable to apply limit bounding"